[WebXR Hit Test] Distinguish gripSpace of an input source to targetRaySpace

fujii opened the pull request:
#### 1802b55bf3640881f6b75e29e39ffcd6596f0f52
<pre>
[WebXR Hit Test] Distinguish gripSpace of an input source to targetRaySpace
<a href="https://bugs.webkit.org/show_bug.cgi?id=303954">https://bugs.webkit.org/show_bug.cgi?id=303954</a>

Reviewed by Sergio Villar Senin.

WebXRInputSpace method nativeOriginInformation() always returned
PlatformXR::InputSourceSpaceType::TargetRay even for gripSpace. It should
return PlatformXR::InputSourceSpaceType::Grip for it.

Test: imported/w3c/web-platform-tests/webxr/hit-test/ar_hittest_subscription_inputSources_gripSpace.https.html

* LayoutTests/imported/w3c/web-platform-tests/webxr/hit-test/ar_hittest_subscription_inputSources_gripSpace.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webxr/hit-test/ar_hittest_subscription_inputSources_gripSpace.https.html: Added.
* Source/WebCore/Modules/webxr/WebXRInputSource.cpp:
(WebCore::WebXRInputSource::update):
* Source/WebCore/Modules/webxr/WebXRInputSpace.cpp:
(WebCore::WebXRInputSpace::nativeOriginInformation const):
* Source/WebCore/Modules/webxr/WebXRInputSpace.h:
(WebCore::WebXRInputSpace::setType):
* Source/WebCore/testing/WebFakeXRDevice.cpp:
(WebCore::SimulatedXRDevice::frameTimerFired):

Canonical link: <a href="https://commits.webkit.org/305117@main">https://commits.webkit.org/305117@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/099e4f1b35b0198b5daa5a17be2f9b5feea50f04

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/137474 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/9834 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/48761 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/145226 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/90447 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/139346 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/10538 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/9961 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/105129 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/76780 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/140419 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7815 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/123212 "Found 1 new API test failure: TestWebKitAPI.WebKit.MediaBufferingPolicy (failure)") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85986 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/7451 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/5168 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/5813 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/116821 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/41366 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/147987 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/9517 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41921 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/113507 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/9535 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/8021 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113852 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/7371 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/119451 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/64152 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/21183 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/9566 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/37504 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/9297 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/73131 "Built successfully") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/9506 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/9358 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

svillar reviewed (approved):
Makes sense. Doesn't fix any test?

fujii committed:
[WebXR Hit Test] Distinguish gripSpace of an input source to targetRaySpace
https://bugs.webkit.org/show_bug.cgi?id=303954

Reviewed by Sergio Villar Senin.

WebXRInputSpace method nativeOriginInformation() always returned
PlatformXR::InputSourceSpaceType::TargetRay even for gripSpace. It should
return PlatformXR::InputSourceSpaceType::Grip for it.

Test: imported/w3c/web-platform-tests/webxr/hit-test/ar_hittest_subscription_inputSources_gripSpace.https.html

* LayoutTests/imported/w3c/web-platform-tests/webxr/hit-test/ar_hittest_subscription_inputSources_gripSpace.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webxr/hit-test/ar_hittest_subscription_inputSources_gripSpace.https.html: Added.
* Source/WebCore/Modules/webxr/WebXRInputSource.cpp:
(WebCore::WebXRInputSource::update):
* Source/WebCore/Modules/webxr/WebXRInputSpace.cpp:
(WebCore::WebXRInputSpace::nativeOriginInformation const):
* Source/WebCore/Modules/webxr/WebXRInputSpace.h:
(WebCore::WebXRInputSpace::setType):
* Source/WebCore/testing/WebFakeXRDevice.cpp:
(WebCore::SimulatedXRDevice::frameTimerFired):

Canonical link: https://commits.webkit.org/305117@main
